### DISCUSSION TITLE
How to combine collapse and swap

### DISCUSSION THREAD

**Discussion by blockhunts** [saadeghi/daisyui#993] [Q&A] [answered]
I want to create a hamburger button that, when it got clicked, it will unhide a list of menu below, and hide it again when it got clicked for the second time. 

Can it be done using combination of swap and collapse component?
All help is appreciated :)

![image](https://user-images.githubusercontent.com/38546608/182297572-5ea1411e-e82c-419a-aa28-9ecd09dd8164.png)

**Comment by saadeghi** [comment, marked as answer]
You need JS to do that.  
Instead of using checkboxes, you can control swap by adding/removing `swap-active` class and you can control `collapse` using `collapse-open`/`collapse-close` class.

**Reply by blockhunts** [reply]
i am sorry but I am still learning laravel. so I am still having a bit difficulty implementing it

**Reply by saadeghi** [reply]
Laravel's PHP code runs on the server. How are you handling front-end functionality using JS?  
Vue? AlpineJS? jQuery? Vanilla JS?

**Reply by blockhunts** [reply]
I used jetstream as the starter template. So it's using AlpineJs and livewire

**Reply by saadeghi** [reply]
so you should be able to use Alpine's [`x-data`](https://alpinejs.dev/directives/data) to define a state (to define if swap is active or not. or if collapse is open or not).  
Then you can [bind class names](https://alpinejs.dev/directives/bind#binding-classes) and change the class names [on click](https://alpinejs.dev/directives/on)

**Reply by blockhunts** [reply]
i see, thank you for the help
